Src/Common/CorrelationIdLookupHelper.cs

```diff
@@ -209,5 +229,56 @@ private string GetExceptionDetailString(Exception ex)
 
             return ex.ToInvariantString();
         }
+
+        /// <summary>
+        /// Structure that represents a failed fetch app Id call.
+        /// </summary>
+        private class FailedResult
+        {
+            /// <summary>
+            /// Initializes a new instance of the <see cref="FailedResult" /> class.
+            /// </summary>
+            /// <param name="failureTime">Time when the failure occurred.</param>
+            /// <param name="failureCode">Failure response code.</param>
+            public FailedResult(DateTime failureTime, HttpStatusCode failureCode)
+            {
+                this.FailureTime = failureTime;
+                this.FailureCode = (int)failureCode;
+            }
+
+            /// <summary>
+            /// Initializes a new instance of the <see cref="FailedResult" /> class.
+            /// </summary>
+            /// <param name="failureTime">Time when the failure occurred.</param>
+            public FailedResult(DateTime failureTime)
+            {
+                this.FailureTime = failureTime;
+
+                // Unknown failure code.
+                this.FailureCode = int.MinValue;
+            }
+
+            /// <summary>
+            /// Gets the time of failure.
+            /// </summary>
+            public DateTime FailureTime { get; private set; }
```

Comment:
Make it a getter-only autoprop. i.e. remove the "private set;" since you only set it in the ctor.